### PR TITLE
Add `social_image_url` to help with the amplification effort

### LINKED ISSUE #24734: Add `social_image_url` values to `build/fileMetadata` per content area
Apply `social_image_url` to the `build/fileMetadata` section of _docfx.json_ file. This is intended to help improve the sharing of content on social media platforms, such as but not limited to; Twitter, LinkedIn, and others who use the `os:image` from an HTML docs `<meta />` tags. We will rely on existing assets that are from the [official .NET brand - here](https://github.com/dotnet/brand).

For example, we might consider applying the image below to all security related content:

.NET security

Or this to IoT content:

.NET IoT

### PR DESCRIPTION
**Pull request by IEvangelist**
## Summary

- Add .NET brand logo to newly created root _/media_ directory
- Create _docfx.json_ `build/fileMetadata/social_image_url` entries that align with images

We may want to add more as they become available and applicable. Also, I'm not entirely certain how these will render as previews &mdash; I have assumptions but if they're wildly inaccurate we can always revert this PR.

Fixes #24734

✔ [.NET Docs &mdash; internal preview](https://review.docs.microsoft.com/en-us/dotnet/?branch=pr-en-us-24861)

Example HTML adds the correct social media `<meta>` tags:
image

### CODE REVIEW

File: docs/media/dot-net-cross-platform.svg [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+<path d="M397.42 105.86L230.35 101.29V105.67L397.42 109.71V105.86Z" fill="#C7E0F4"/>
+<path d="M365.62 199.74L267.86 203.43C267.05 203.47 266.39 204.25 266.39 205.17V212.88L366.89 208.37V201.23C366.89 200.37 366.33 199.71 365.62 199.74Z" fill="#3085E9"/>
+<path d="M295.68 209.5L292.41 209.67C292.03 209.69 291.73 209.35 291.73 208.92V205.13C291.73 204.7 292.03 204.33 292.41 204.31L295.68 204.15C296.05 204.13 296.36 204.47 296.36 204.9V208.67C296.35 209.11 296.05 209.48 295.68 209.5Z" fill="#8DBCF4"/>
+<path d="M286.72 207.39C286.72 208.96 285.61 210.3 284.23 210.37C282.85 210.44 281.73 209.22 281.73 207.64C281.73 206.06 282.85 204.72 284.23 204.66C285.6 204.59 286.72 205.81 286.72 207.39Z" fill="#8DBCF4"/>
+<path d="M306.34 206.26C306.34 207.81 305.26 209.13 303.93 209.2C302.59 209.27 301.51 208.06 301.51 206.51C301.51 204.96 302.6 203.64 303.93 203.57C305.26 203.5 306.34 204.71 306.34 206.26Z" fill="#8DBCF4"/>
+<path d="M316.26 205.9C316.26 207.44 315.2 208.74 313.88 208.81C312.56 208.88 311.49 207.68 311.49 206.14C311.49 204.6 312.56 203.29 313.88 203.23C315.19 203.16 316.26 204.36 316.26 205.9Z" fill="#8DBCF4"/>
```
Comment (CamSoper) [inline docs/media/dot-net-cross-platform.svg:41]: This line looks wrong.

**Review by CamSoper — APPROVED** [review]
I 💜 this idea!

File: docs/media/dot-net-cross-platform.svg (hunk 1 above)
Comment (CamSoper) [reply docs/media/dot-net-cross-platform.svg:41]: What? You find it unlikely that I could decipher SVG markup in my head? 😉

Comment (IEvangelist) [reply docs/media/dot-net-cross-platform.svg:41]: How so? It came from the .NET brand repo, so if it's wrong here - it's wrong there as well. And the rendering looked good, no?

**Pull request merged**